e2e-tests/src/localstack.rs

```diff
@@ -0,0 +1,219 @@
+
+        // getting the PIE file from s3 bucket using URL provided
+        let file = reqwest::get(format!(
+            "https://madara-orchestrator-sharp-pie.s3.amazonaws.com/{}-SN.zip",
+            self.l2_block_number
+        ))
+        .await?;
+        let file_bytes = file.bytes().await?;
+
+        // putting the pie file into localstack s3
+        let s3_file_key = self.l2_block_number.to_string() + "/pie.zip";
+        s3_client.put_data(file_bytes, &s3_file_key).await?;
+        println!("✔️ PIE file uploaded to localstack s3");
+
+        Ok(())
+    }
+
+    /// Event Bridge setup
+    pub async fn setup_event_bridge(&self, worker_trigger_type: WorkerTriggerType) -> color_eyre::Result<()> {
+        let event_bridge_client = self.event_bridge_client().await;
+        let sqs_client = self.sqs_client().await;
+
+        let rule_name = "worker_trigger_scheduled";
+
+        event_bridge_client
+            .put_rule()
+            .name(rule_name)
+            .schedule_expression("rate(1 minute)")
+            .state(RuleState::Enabled)
+            .send()
+            .await?;
+        let queue_url = sqs_client.get_queue_url().queue_name(WORKER_TRIGGER_QUEUE).send().await?;
+
+        let queue_attributes = sqs_client
+            .get_queue_attributes()
+            .queue_url(queue_url.queue_url.unwrap())
+            .attribute_names(QueueAttributeName::QueueArn)
+            .send()
+            .await?;
+        let queue_arn = queue_attributes.attributes().unwrap().get(&QueueAttributeName::QueueArn).unwrap();
+
+        // Create a sample WorkerTriggerMessage
+        let message = WorkerTriggerMessage { worker: worker_trigger_type.clone() };
+        let event_detail = serde_json::to_string(&message)?;
+
+        // Create the EventBridge target with the input transformer
+        let input_transformer = InputTransformer::builder()
+            .input_paths_map("$.time", "time")
+            .input_template(event_detail.to_string())
+            .build()?;
+
+        event_bridge_client
+            .put_targets()
+            .rule(rule_name)
+            .targets(
+                Target::builder()
+                    .id(format!("worker-trigger-target-{:?}", worker_trigger_type))
+                    .arn(queue_arn)
+                    .input_transformer(input_transformer)
+                    .build()?,
+            )
+            .send()
+            .await?;
+
+        println!("🌉 Event bridge setup completed. Trigger Type : {:?}", worker_trigger_type);
+
+        Ok(())
+    }
+
+    /// Generic function to send message to any of the queues
+    pub async fn send_message_to_queue(&self, queue_url: &str, message_body: &str) -> color_eyre::Result<()> {
+        let sqs_client = self.sqs_client().await;
+        sqs_client.send_message().queue_url(queue_url).message_body(message_body).send().await?;
+        Ok(())
+    }
+
+    async fn sqs_client(&self) -> aws_sdk_sqs::Client {
```

Comment:
the clients should be a part of the `LocalStack` struct + the new method which builds the client and returns the struct